Tools/ardupilotwaf/cxx_checks.py

```diff
@@ -173,3 +173,11 @@ def check_libiio(cfg, env):
     if cfg.env['LIB_LIBIIO']:
         env.LIB += ['dl']
     return True
+
+@conf
+def check_libdl(cfg, env):
+    ret = cfg.check(compiler='cxx', lib='dl')
+    if ret:
+        env.LIB += cfg.env['LIB_DL']
+        cfg.define('HAVE_LIBDL', 1)
+    return ret
```

Comment:
most of the other ones use pkg-config, which isn't the right method for libdl I think
I've fixed the mandatory libdl now